packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -16,361 +8,325 @@ import {
-    if (colorCells.length < 1 || columnCount < 1) {
-      return null;
-    }
-    return (
-      <Grid
-        {...this.props}
-        id={this._id}
-        items={this._getItemsWithIndex(colorCells)}
-        columnCount={columnCount}
-        onRenderItem={this._renderOption}
-        ariaPosInSet={ariaPosInSet}
-        ariaSetSize={ariaSetSize}
-        shouldFocusCircularNavigate={shouldFocusCircularNavigate}
-        doNotContainWithinFocusZone={doNotContainWithinFocusZone}
-        onBlur={this._onSwatchColorPickerBlur}
-        theme={this.props.theme!}
-        styles={{
-          root: classNames.root,
-          tableCell: classNames.tableCell,
-          focusedContainer: classNames.focusedContainer,
-        }}
-      />
-    );
-  }
+  const gridStyles = {
+    root: classNames.root,
+    tableCell: classNames.tableCell,
+    focusedContainer: classNames.focusedContainer,
+  };
 
   /**
    * When the whole swatchColorPicker is blurred,
    * make sure to clear the pending focused stated
    */
-  private _onSwatchColorPickerBlur = (): void => {
-    if (this.props.onCellFocused) {
-      this._cellFocused = false;
-      this.props.onCellFocused();
+  const onSwatchColorPickerBlur = React.useCallback((): void => {
+    if (onCellFocused) {
+      internalState.cellFocused = false;
+      onCellFocused();
     }
-  };
-
-  /**
-   * Get the selected item's index
-   * @param items - The items to search
-   * @param selectedId - The selected item's id to find
-   * @returns - The index of the selected item's id, -1 if there was no match
-   */
-  private _getSelectedIndex(items: IColorCellProps[], selectedId: string): number | undefined {
-    const selectedIndex = findIndex(items, item => item.id === selectedId);
-    return selectedIndex >= 0 ? selectedIndex : undefined;
-  }
-
-  /**
-   * Render a color cell
-   * @param item - The item to render
-   * @returns - Element representing the item
-   */
-  private _renderOption = (item: IColorCellProps): JSX.Element => {
-    const props = this.props;
-    const id = this._id;
-
-    return (
-      <ColorPickerGridCell
-        item={item}
-        idPrefix={id}
-        color={item.color}
-        styles={props.getColorGridCellStyles}
-        disabled={props.disabled}
-        onClick={this._onCellClick}
-        onHover={this._onGridCellHovered}
-        onFocus={this._onGridCellFocused}
-        selected={this.state.selectedIndex !== undefined && this.state.selectedIndex === item.index}
-        circle={props.cellShape === 'circle'}
-        label={item.label}
-        onMouseEnter={this._onMouseEnter}
-        onMouseMove={this._onMouseMove}
-        onMouseLeave={this._onMouseLeave}
-        onWheel={this._onWheel}
-        onKeyDown={this._onKeyDown}
-        height={props.cellHeight}
-        width={props.cellWidth}
-        borderWidth={props.cellBorderWidth}
-      />
-    );
-  };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [onCellFocused]);
```

Comment:
In this case it's probably better to just add `internalState` to the dep list (even though we know it won't change) so that if anybody edits this in the future and adds references to more things which *might* change, the rule will catch it.
```suggestion
  }, [onCellFocused, internalState]);
```